shadowsocks/dialer_test.go

```diff
@@ -0,0 +1,212 @@
+package shadowsocks
```

Comment:
We should add a benchmark. It's very easy to get regressions in performance without noticing.

Reply:
Done. Added one for each TCP and UDP.